app/services/flow/base_step.rb

```diff
@@ -32,6 +32,7 @@ def form_submit
 
     def failure(message, extra = {})
       flow_session[:error_message] = message
+      extra ||= {}
```

Comment:
Instead of instantiating this, would it be worth tracking down the clients and mocks to make sure they return something here? That way the API they implement won't drift.

Reply:
Hmm ...could you explain?  I'm trying to handle the cases where they pass nothing (which should be a majority of the time) or when they pass in nil

Reply:
I guess i could simply disallow nil

Reply:
That's what I mean, where are we returning nil? Could we returned a hash there instead?

Reply:
Actually this might have been an old error from my first push.  Removing it now

Reply:
switched it to def failure(message, extra = nil)